packages/cache/src/cache.ts

```diff
@@ -182,17 +172,41 @@ export async function saveCache(
       await listTar(archivePath, compressionMethod)
     }
 
-    const fileSizeLimit = 10 * 1024 * 1024 * 1024 // 10GB per repo limit
     const archiveFileSize = utils.getArchiveFileSizeInBytes(archivePath)
     core.debug(`File Size: ${archiveFileSize}`)
-    if (archiveFileSize > fileSizeLimit) {
-      throw new Error(
-        `Cache size of ~${Math.round(
-          archiveFileSize / (1024 * 1024)
-        )} MB (${archiveFileSize} B) is over the 10GB limit, not saving cache.`
+
+    const cacheSize = archiveFileSize
+    core.debug('Reserving Cache')
+    const reserveCacheResponse = await cacheHttpClient.reserveCache(
+      key,
+      paths,
+      {
+        compressionMethod,
+        cacheSize
+      }
+    )
+
+    if (
+      reserveCacheResponse?.statusCode === 400 &&
+      reserveCacheResponse?.typeKey === 'InvalidReserveCacheRequestException'
```

Comment:
Why handle this exception explicitly? IMO just check for any non null `error` and if so, throw a ReserveCacheError and use same message from `error.message`

Reply:
Using error.message is it exists else using a default message with similar information.